compiler/dyno/lib/parsing/parsing-queries.cpp

```diff
@@ -164,26 +164,144 @@ const ModuleVec& parse(Context* context, UniqueString path) {
   return QUERY_END(result);
 }
 
-static const std::vector<std::string>&
+static const std::vector<UniqueString>&
 moduleSearchPathQuery(Context* context) {
   QUERY_BEGIN_INPUT(moduleSearchPathQuery, context);
 
   // return the empty path if it wasn't already set in
   // setModuleSearchPath
-  std::vector<std::string> result;
+  std::vector<UniqueString> result;
 
   return QUERY_END(result);
 }
 
-const std::vector<std::string>& moduleSearchPath(Context* context) {
+const std::vector<UniqueString>& moduleSearchPath(Context* context) {
   return moduleSearchPathQuery(context);
 }
 
 void setModuleSearchPath(Context* context,
-                         std::vector<std::string> searchPath) {
+                         std::vector<UniqueString> searchPath) {
   QUERY_STORE_INPUT_RESULT(moduleSearchPathQuery, context, searchPath);
 }
 
+
+static const UniqueString&
+internalModulePathQuery(Context* context) {
+  QUERY_BEGIN_INPUT(internalModulePathQuery, context);
+
+  // use empty string if wasn't already set by setInternalModulePath
+  UniqueString result;
+
+  return QUERY_END(result);
+}
+
+UniqueString internalModulePath(Context* context) {
+  return internalModulePathQuery(context);
+}
+
+void setInternalModulePath(Context* context, UniqueString path) {
+  QUERY_STORE_INPUT_RESULT(internalModulePathQuery, context, path);
+}
+
+static const UniqueString&
+bundledModulePathQuery(Context* context) {
+  QUERY_BEGIN_INPUT(bundledModulePathQuery, context);
+
+  // use empty string if wasn't already set by setInternalModulePath
+  UniqueString result;
+
+  return QUERY_END(result);
+}
+
+UniqueString bundledModulePath(Context* context) {
+  return bundledModulePathQuery(context);
+}
+void setBundledModulePath(Context* context, UniqueString path) {
+  QUERY_STORE_INPUT_RESULT(bundledModulePathQuery, context, path);
+}
+
+void setupModuleSearchPaths(Context* context,
+                            const std::string& chpl_home,
```

Comment:
I understand the intention here, but having some names be snake case personally threw me out of the code.